File: judge/tests/bots/rps_rock.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())

		if line == "START" {
			fmt.Println("ROCK")
		} else if strings.HasPrefix(line, "ROUND") {
			fmt.Println("ROCK")
		} else if strings.HasPrefix(line, "SCORE") {
			continue
		} else if line == "END" {
			break
		}
	}
}

File: judge/tests/bots/rps_rock.c
#include <stdio.h>
#include <string.h>

int main() {
    char line[256];

    while (fgets(line, sizeof(line), stdin)) {
        // Remove newline
        line[strcspn(line, "\n")] = 0;

        if (strcmp(line, "START") == 0) {
            printf("ROCK\n");
            fflush(stdout);
        } else if (strncmp(line, "ROUND", 5) == 0) {
            printf("ROCK\n");
            fflush(stdout);
        } else if (strncmp(line, "SCORE", 5) == 0) {
            continue;
        } else if (strcmp(line, "END") == 0) {
            break;
        }
    }

    return 0;
}
